src/Payments.sol

```diff
@@ -1804,15 +1805,39 @@ contract Payments is ReentrancyGuard {
         auction.startPrice = uint88(auctionPrice);
         auction.startTime = uint168(block.timestamp);
 
-        (bool success,) = BURN_ADDRESS.call{value: msg.value}("");
-        require(success, Errors.NativeTransferFailed(BURN_ADDRESS, msg.value));
+        burnViaPrecompile(msg.value);
 
         {
             uint256 actual = transferOut(token, recipient, requested);
             // handle fee-on-transfer and hidden-denominator tokens
             fees.funds = available - actual;
         }
     }
+
+    /**
+     * @dev Burns FIL using the FVM precompile to call the burn actor (ID 99)
+     * @param amount The amount of FIL to burn
+     */
+    function burnViaPrecompile(uint256 amount) private {
+        // Using simpler encoding to reduce stack usage
+        bytes memory empty = "";
+        bytes memory data = abi.encode(
+            uint64(0), // method 0
+            amount, // value
+            uint64(0), // flags
+            uint64(0), // codec
+            empty, // params
```

Comment:
Since this is an empty buffer, there is an encoding trick you can do where you point this to another zero you already have in the calldata. For example, because your `method` is 0, you can put `0` for the `params`. This encoding uses slightly less data.

Reply:
8e09330